eight/eight_nieoh.py

```diff
@@ -0,0 +1,35 @@
+class CircularQueue:
+
+    def __init__(self, n):
+        self.len = n
+        self.size = 0
+        self.circular_queue = [0] * n
+        self.start = 0
+        self.end = 0
+
+    def enqueue(self, data):
+        self.size += 1
+        if self.size > self.len:
+            self.resize(self.len)
+        self.circular_queue[self.end] = data
+        self.end = (self.end + 1)%self.len
+
+    def dequeue(self):
+        deq = self.circular_queue[self.start]
+        self.size -= 1
+        self.start = (self.start + 1)%self.len
+        return deq
+
+    def num_elts(self):
+        return self.size
+   
+    def resize(self, num):
+        temp = [0] * num
+        timp = self.start
+        for i in range(self.len):
```

Comment:
Consider using dequeue here.

Reply:
You mean instead of taking circular_queue[index]?

Reply:
Yeah. Not so sure any more, but you might be able to do 
```python
new_one = Queue(capacity * 2)
for x in old_one.dequeue():
 new_one.enqueue(x)
```
and then steal the list / other data out of `new_one`.

Not sure if that's better or not though.